lib/signature/gen_sign.dart

```diff
@@ -1,16 +1,36 @@
+import 'dart:isolate';
+
 import 'package:path_provider/path_provider.dart';
 import 'package:fexr/signature/dependencies.dart';
 
-class GenerateSign {
- Future<String> genSignFromShares(String imagePath, String hash) async{
+class SignInput {
+  String imagePath;
+  String hash;
+  SignInput({required this.imagePath, required this.hash});
+}
+
+Future<void> genSignFromSharesIsolate(List<dynamic> args) async{
+    var sendPort = args[0] as SendPort;
+    var input = args[1] as SignInput;
+    var hash = input.hash;
+    var imagePath = input.imagePath;
     Future<String> firstPrivatebin = Dependencies().imageToBinary(imagePath);
     String firstPrivate = await firstPrivatebin;
     List<String> privateIntegerArrayString = List<String>.generate(firstPrivate.length, (index) => firstPrivate[index]);
     List<int> privateIntegerArray = privateIntegerArrayString.map((index) => int.parse(index)).toList();
     Map P = Dependencies().randomPositions("signer", hash, 32, privateIntegerArray);
     var finalPos = P["posForSign"];
     List<int> p1Sign = Dependencies().getPrivatePosition(finalPos, privateIntegerArray);
-    return p1Sign.join("");
+    sendPort.send(p1Sign.join(""));
+}
+
+class GenerateSign {
+
```

Comment:
extra new line

Reply:
Thanks for pointing that out. Fixing now.